lib/modules/manager/npm/post-update/index.ts

```diff
@@ -241,6 +241,27 @@ export async function writeExistingFiles(
     if (packageFile.pnpmShrinkwrap && config.reuseLockFiles === false) {
       await deleteLocalFile(packageFile.pnpmShrinkwrap);
     }
+
+    const yarnrcFilename = upath.join(basedir, '.yarnrc');
```

Comment:
Can't this logic be in a yarn-specific section

Reply:
That's a good idea. Since I don't know the codebase that well I would appreciate some assistance in this. Where do you think is appropriate?

Here are some alternatives I can think of:

- New function `massageYarnrc` within `index.ts`
- New function `massageYarnrc` in `yarn.ts`
- Modify `checkYarnrc` in `yarn.ts`

Reply:
I think the latter because you can see it already writes out a modified yarnrc here: https://github.com/renovatebot/renovate/blob/4d8e2c704f15b37140f671601cf361c931d28421/lib/modules/manager/npm/post-update/yarn.ts#L64-L67

Reply:
Have fixed this now, can you take a look again?

I also ran it against my repro repo again and it seems to work:
https://github.com/aommm/renovate_reproduction_11356/pull/4